src/pages/AddItem/index.js

```diff
@@ -20,27 +24,41 @@ export const AddItemPage = () => {
   const onSubmit = (event) => {
     event.preventDefault();
     setIsLoading(true);
-
-    addProduct(formState)
-      .then((res) => console.log(res))
-      .catch(() => setError(true))
-      .finally(() => {
-        setIsLoading(false);
-
-        setFormState({
-          productName: '',
-          timeFrame: '7',
-          lastPurchaseDate: null,
+    console.log(products);
+    /*
+    Checks the state if the product already exists
+    Find returns undefined if it has not matches
+    */
+    const normalizer = (input) => input.toUpperCase().replaceAll(/[.,:;]/g, '');
+    const normalizedInput = normalizer(formState.productName);
+    const checkProduct = products.find(
+      (product) => normalizer(product.productName) === normalizedInput,
+    );
+    /*
+    We have to compare explicitely if it is undefined, otherwise
+    it wouldn't work as expected
+    */
+    if (checkProduct === undefined) {
+      addProduct(formState)
+        .then((res) => console.log(res))
+        .finally(() => {
+          setIsLoading(false);
+          console.log(formState.productName);
+          setFormState({
+            productName: '',
+            timeFrame: '7',
+            lastPurchaseDate: null,
+          });
         });
+    }
+    setIsLoading(false);
+    setMessage(
+      checkProduct
+        ? 'It seems you have already added this product. Try another one.'
+        : 'Successfully created product!',
+    );
```

Comment:
I found the deal interesting if the product already exists on the list. But I don't know if it's a good idea to take the `catch error` from the product addition promise. It could be that an error occurs outside of our control, in Firebase for example, and I believe the user needs to be informed of these possible errors as well. What do you think? Would it be possible to adjust the code to be able to handle all cases?

```
if (checkProduct === undefined) {
  addProduct(formState)
    .then((res) => console.log(res))
    .catch(() => setMessage('Unable to add product. Please, try again.'))
    .finally(() => {
      /****/
      if (!message) {
        setMessage('Successfully created product!')
      } 
    })
} else {
  setMessage('It seems you have already added this product. Try another one.')
}

setIsLoading(false);
setTimeout(() => setMessage(null), 3000);
```

Reply:
Finally, I know this file gave us a lot of problems, but I believe it could be interesting an `else` statement. It could be easier to read. Well, it's only a suggestion, not something mandatory.